kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -93,6 +100,68 @@ public static StructType convertToPhysicalSchema(
         }
     }
 
+    /** Returns the physical name for a given {@link StructField} */
+    public static String getPhysicalName(StructField field) {
+        if (field.getMetadata().contains(COLUMN_MAPPING_PHYSICAL_NAME_KEY)) {
+            return (String) field
+                    .getMetadata()
+                    .get(COLUMN_MAPPING_PHYSICAL_NAME_KEY);
```

Comment:
did you create that issue about `getString` on `FieldMetadata` class? we can leave a TODO here

Reply:
based on https://github.com/delta-io/delta/pull/3393#discussion_r1684608738 I was waiting for a response whether we should do it, so I haven't opened an issue yet

Reply:
I've opened https://github.com/delta-io/delta/issues/3419 to address that